memory-fs/memory-fs.go

```diff
@@ -22,8 +26,105 @@ type openFS interface {
 	FS() fs.File
 }
 
-func New() MemoryFS                               { panic("not implemented") }
-func Fork(name string) (MemoryFS, error)          { panic("not implemented") }
-func Persistence(mfs MemoryFS, name string) error { panic("not implemented") }
+func New() MemoryFS {
+	return &mfs{
+		root: &dir{
+			RWMutex: sync.RWMutex{},
+			name:    "/",
+			modTime: time.Now(),
+			node:    make(map[string]openFS),
+		},
+	}
+}
+
+func Fork(name string) (MemoryFS, error) {
```

Comment:
You may be mistaken.
e.g /data/files
What we need to read is the file under /data/files